### PR TITLE
[lldb] Fix use-after-free from recent SP refactor found by ASAN

### PR DESCRIPTION
**Pull request by adrian-prantl**

### CODE REVIEW

**Comment by adrian-prantl** [comment]
@swift-ci test

**Comment by adrian-prantl** [comment]
@swift-ci test

File: lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.h [hunk 1]
```diff
@@ -907,7 +908,7 @@ class SwiftASTContext : public TypeSystemSwift {
   /// Data members.
   /// @{
   // Always non-null outside of unit tests.
```
Comment (felipepiovezan) [inline lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.h]: Should we also remove this comment?

**Review by felipepiovezan — APPROVED** [review]
LGTM, left one minor comment

**Review by JDevlieghere — APPROVED** [review]

**Comment by adrian-prantl** [comment]
@swift-ci test

**Pull request merged**